src/mesido/electricity_physics_mixin.py

```diff
@@ -114,6 +119,26 @@ def pre(self):
                 self.__asset_is_switched_on_var[var_name] = ca.MX.sym(var_name)
                 self.__asset_is_switched_on_bounds[var_name] = (0.0, 1.0)
 
+        if options["electrolyzer_efficiency"] == ElectrolyzerOption.LINEARIZED_THREE_LINES_EQUALITY:
+            n_lines = 3
+            for n_line in range(n_lines):
+                self.__electrolyzer_is_active_linear_segment_map[f"line_{n_line}"] = {}
+                self.__electrolyzer_is_active_linear_segment_var[f"line_{n_line}"] = {}
+                self.__electrolyzer_is_active_linear_segment_bounds[f"line_{n_line}"] = {}
+                for asset in [
+                    *self.energy_system_components.get("electrolyzer", []),
```

Comment:
It is preferred to have a different order in the loops, first loop over the assets and then over the lines, such that one can easily check if an asset is in this map/if it contains these variables.

Reply:
Changed